### PR TITLE
fix (web): panorama / 360 view - partial panorama: use photo-sphere-viewer

### PR DESCRIPTION
**Pull request by dmitry-brazhenko**
Previous PR: https://github.com/immich-app/immich/pull/3412

This PR should fix 2 issues: https://github.com/immich-app/immich/issues/3490 and https://github.com/immich-app/immich/issues/3465

I changed 360 images viewer. Now I use photo-sphere-viewer. 

I checked "cropped/partial" panoramas and they work fine. 

image

### CODE REVIEW

File: web/src/lib/components/asset-viewer/photo-sphere-viewer-adapter.svelte [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+<div bind:this={container} />
+
+<style>
+  div {
+    width: 100%;
+    height: 100%;
+    margin-bottom: 0;
+  }
+</style>
```
Comment (martabal) [inline web/src/lib/components/asset-viewer/photo-sphere-viewer-adapter.svelte:35]: Can you use 
```html
<div class="h-full w-full mb-0" bind:this={container} />
``` 
instead ?

**Review by danieldietzler — APPROVED** [review]

**Comment by dmitry-brazhenko** [comment]
@danieldietzler  could you help me to merge this PR?

**Comment by danieldietzler** [comment]
> @danieldietzler could you help me to merge this PR?

I don't merge PRs, only some people do :)
Why is it so urgent? It's not like we'll do a release right after this PR

**Comment by dmitry-brazhenko** [comment]
No urgency, just wanted not to forget about this :)

**Comment by danieldietzler** [comment]
We usually don't "forget" pull requests, no worries.

**Comment by mertalev** [comment]
The photo-sphere-viewer documentation says that it handles cropped/partial panoramas by looking at image metadata. We strip most metadata from preview images now, so can you confirm that a newly uploaded partial panorama still works?

**Review by alextran1502 — APPROVED** [review]

**Comment by alextran1502** [comment]
@mertalev tested on panorama taken on iPhone and it works

**Comment by dmitry-brazhenko** [comment]
Yep, I also tested on newly uploaded images. As far as I see, exif is not removed from image

**Comment by rovo89** [comment]
@dmitry-brazhenko I think panorama-viewer.css is now unused, isn't it? Then I guess it should be deleted.

**Comment by dmitry-brazhenko** [comment]
@rovo89  thanks a lot for your attentiveness!!

Here is a PR: https://github.com/immich-app/immich/pull/7012

**Comment by waclaw66** [comment]
A few comments...
- panorama photos got attributes `InitialViewHeadingDegrees` and `InitialViewPitchDegrees`, ignoring them causes that image is rotated to the wrong side initially, they should be kept and set to `poseHeading` and `posePitch` panorama viewer properties
- there is a white vertical line between info panel and panorama, it's because viewport of panorama has partial dimensions, but the viewer uses integer (truncated) ones
 
obrazek

**Comment by dmitry-brazhenko** [comment]
That's good point. I created an issue: https://github.com/immich-app/immich/issues/7243

**Pull request merged**